LibGAP.jl/src/julia_to_gap.jl

```diff
@@ -0,0 +1,99 @@
+## Converters
+
+## Default
+julia_to_gap(x::GAPInputType, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = x
+
+
+## Integers
+julia_to_gap(x::Int128, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = MakeObjInt(BigInt(x)) # FIXME: inefficient hack
+#julia_to_gap(x::Int64)  = x
+julia_to_gap(x::Int32, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = Int64(x)
+julia_to_gap(x::Int16, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = Int64(x)
+julia_to_gap(x::Int8, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = Int64(x)
+
+## Unsigned Integers
+julia_to_gap(x::UInt128, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = MakeObjInt(BigInt(x)) # FIXME: inefficient hack
+julia_to_gap(x::UInt64, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = MakeObjInt(BigInt(x)) # FIXME: inefficient hack
+julia_to_gap(x::UInt32, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = Int64(x)
+julia_to_gap(x::UInt16, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = Int64(x)
+julia_to_gap(x::UInt8, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive  = Int64(x)
+
+## BigInts
+julia_to_gap(x::BigInt, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive = MakeObjInt(x)
+
+## Rationals
+function julia_to_gap(x::Rational{T}, ::Val{Recursive}=Val(false), recursion_dict = nothing) where Recursive where T <: Integer
+    denom_julia = denominator(x)
+    numer_julia = numerator(x)
+    if denom_julia == 0
+        if numer_julia >= 0
+            return GAP.Globals.infinity
```

Comment:
Should there perhaps also be a reverse conversion for this? But of course that's difficult, as on the GAP side, `infinity` and `-infinity` are objects, and "only" in the filter `IsCyc`, not in `IsRat`. Huh :/

Reply:
@fingolfin As far as I remember,
the object `infinity` had originally been designed just for having a natural object
that is larger than any integer, and that can be compared with integers via `=` and `<`.
In particular, there were no arithmetic operations for `infinity`,
and also `-infinity` did not yet exist.
Because of the wish to compare the new object `infinity` with integers,
the family `CyclotomicsFamily` was a natural choice for it,
and thus `infinity` is in the filter `IsCyclotomic` --but not in `IsCyc`, `IsRat`, or `IsInt`.

Before the introduction of `infinity`, other GAP objects of different types had been used
for this kind of comparison, for example `E(3)`,
assuming that GAP regards them as larger than integers.
Using `infinity` in such situations is nicer because it is self-explanatory;
besides that, I do not think that `infinity` is useful.

Reply:
I am not sure how this is relevant for the discussion at hand, though?

BTW, I added `-infinity`, and I find it and `infinity` quite useful for all kinds of things, e.g. for polynomial degree functions, or arbitrary valuations; as default values for minimum / maximum operations; etc.. But whether one finds them useful or not is kind of irrelevant for this discussion. What we have to deal with is issue #162. We can either throw an error if we encounter the Julia rationals 1//0 and -1//0; or convert them to infinity and -infinity (as this PR does). In the latter case, the question arises whether we want to make the conversion roundtrip or not. While in general, I am in favor for clean roundtrip conversions, but is of course not always possible; but then perhaps there should be a comment here pointing that out.

Reply:
I think it is sensible to convert to `infinity` here.

Remember, we are talking about a very specific corner-case here, Julia rationals with `0` as denominator. I do not think it is that important anyway.